src/quickgui/plugin/editor/qgsquickvaluemap.qml

```diff
@@ -25,35 +25,63 @@ import QgsQuick 0.1 as QgsQuick
  */
 Item {
   signal valueChanged(var value, bool isNull)
+  id: fieldItem
 
   anchors {
     left: parent.left
     right: parent.right
     rightMargin: 10 * QgsQuick.Utils.dp
   }
 
-  height: childrenRect.height + 10 * QgsQuick.Utils.dp
+  height: customStyle.height
 
   ComboBox {
     id: comboBox
 
     property var reverseConfig: ({})
     property var currentValue: value
-
+    property var currentMap
+    property var currentKey
+    height: parent.height
     anchors { left: parent.left; right: parent.right }
-
     currentIndex: find(reverseConfig[value])
 
+    ListModel {
+        id: listModel
+    }
+
     Component.onCompleted: {
-      model = Object.keys(config['map']);
-      for(var key in config['map']) {
-        reverseConfig[config['map'][key]] = key;
+      if( config['map'] )
+      {
+        if( config['map'].length )
+        {
+          //it's a list (>=QGIS3.0)
+          for(var i=0; i<config['map'].length; i++)
+          {
+            currentMap = config['map'][i]
+            currentKey = Object.keys(currentMap)[0]
+            listModel.append( { text: currentKey } )
+            reverseConfig[currentMap[currentKey]] = currentKey;
+          }
+          model=listModel
+          textRole = 'text'
+        }
+        else
+        {
+          //it's a map (<=QGIS2.18)
```

Comment:
Do we need to support 2.18?

Reply:
Its ported from QField and I would keep it.